### PR TITLE
network: Notify users when the same address is configured multiple times

### LINKED ISSUE #20891: PreferredLifetime = 0 isn't setting the lifetime to 0
**systemd version the issue has been seen with**
 > 249.4

I've been using the config below:

```
$ cat /etc/systemd/network/1-eth0.network 

[Network]
Address = <primary address>
Address = <secondary address>
IPv6AcceptRA = true

[Match]
MACAddress = <mac address>
Name = eth0

[IPv6AcceptRA]
UseAutonomousPrefix = false
UseOnLinkPrefix = false
DHCPv6Client = false

[Address]
Address = <secondary address>
PreferredLifetime = 0
```

If I do `ip addr` and the secondary address has preferred lifetime set to forever, running `networkctl reload && networkctl reconfigure eth0` on systemd 248.5 would change the lifetime to 0. However in systemd 249.4 this isn't changing to 0.

### PR DESCRIPTION
**Pull request by daandemeyer**
cc @yuwata @anitazha 

Fixes #20891

### CODE REVIEW

File: src/network/networkd-address.c [hunk 1]
```diff
@@ -1400,6 +1401,28 @@ int manager_rtnl_process_address(sd_netlink *rtnl, sd_netlink_message *message,
...
 
+static int maybe_warn_address_already_configured(
+                Network *network,
+                const Address *address,
+                const char *unit,
+                const char *filename,
+                unsigned int line) {
+        Address *a;
+        _cleanup_free_ char *addr = NULL;
+
```
Comment (bluca) [inline src/network/networkd-address.c:1412]: assert on at least address and network

File: src/network/networkd-address.c [hunk 2]
```diff
@@ -1400,6 +1401,28 @@ int manager_rtnl_process_address(sd_netlink *rtnl, sd_netlink_message *message,
         return 1;
 }
 
+static int maybe_warn_address_already_configured(
+                Network *network,
+                const Address *address,
+                const char *unit,
+                const char *filename,
+                unsigned int line) {
+        Address *a;
```
Comment (bluca) [inline src/network/networkd-address.c:1410]: empty new line before first declaration

**Comment by yuwata** [comment]
Address settings should be deduped in `network_drop_invalid_address_section()`. See #20892.

**Comment by DaanDeMeyer** [comment]
Let's close in favor of #20892

**Pull request closed without merging**